src/main/java/org/openpnp/vision/pipeline/stages/FilterRects.java

```diff
@@ -0,0 +1,253 @@
+    public Result process(CvPipeline pipeline) throws Exception {
+        if (rotatedRectsStageName == null) {
+            return null;
+        }
+        
+        Result result = pipeline.getResult(rotatedRectsStageName);
+        if (result == null || result.model == null) {
+            return null;
+        }
+        
+        List<RotatedRect> rects; 
+        if (result.model instanceof RotatedRect) {
+            rects = Collections.singletonList((RotatedRect)result.model);
+        }
+        else {
+            rects = (List<RotatedRect>)result.model;
+        }
+
+        List<RotatedRect> results = new ArrayList<RotatedRect>();
+                
+        double wmax, wmin, lmax, lmin, armax, armin, rw, rl, rar;
+        int sizeType = 0;
+        String pass;
+        boolean wok, lok, arok;
+        
+        // only positive dimensions
+        wmax = Math.max(widthMax, widthMin);
+        wmin = Math.min(widthMax, widthMin);
+        lmax = Math.max(lengthMax, lengthMin);
+        lmin = Math.min(lengthMax, lengthMin);
+        armax = Math.max(aspectRatioMax, aspectRatioMin);
+        armin = Math.min(aspectRatioMax, aspectRatioMin);
+        
+        // check input parameters
+        if (armin == 0 && (wmax == 0 || lmax == 0))
+        {
+          // cannot accept this as there will be divisions by 0 
+          throw new Exception("If width or length limits are 0, then both aspectRatioMax and aspepctRatioMin must be non zero.");            
+        }
+        if (armax != 0 && (wmax == 0 && lmax == 0))
+        {
+          sizeType = 2;
+        }
+        else if (lmax == 0) 
+        {
+          // derive length from aspect ratio
+          lmax = wmax / armin; lmin = wmin / armax;
+          sizeType = 1;
+        } 
+        else if (wmax == 0)
+        {
+          wmax = lmax * armax; wmin = lmin * armin;
+          sizeType = 1;
+        }
+        else {
+          sizeType = 3;
+        }
+        if (enableLogging) {
+          if (sizeType == 1)
+            Logger.info("Deriving "+(wmax == 0? "width":"length")+ " limits based on aspect ratio limits.");
+          else if (sizeType == 2)
+            Logger.info("Pass any size rect within aspect ratio limits.");
+        }
+        
+        // iterate over input rects
+        for (RotatedRect rect : rects) {
+        
+          rw = Math.max(rect.size.width,rect.size.height);
+          rl = Math.min(rect.size.width,rect.size.height);
+          // ignore aspect ratio = NaN or infinity
+          rar= rw / rl;
+          if (Double.isNaN(rar) || Double.isInfinite(rar))
+            continue;
+          
+          if (sizeType == 2) {
+            // Any size:
+            // Select rects based only on the aspect ratio of each input rectangle
+            lmin = rl;
+            lmax = rl;
+            wmax = rl *armax; 
+            wmin = rl * armin;
+          }
+
+          // check criteria
+          if (rw >= wmin && rw <= wmax) wok = true;
```

Comment:
Please see https://github.com/openpnp/openpnp/wiki/Developers-Guide#coding-style with regards to use of optional braces. There's several instances of that in this file, so I won't flag each one, but please fix them all.